benchmarks/filtsmooth.py

```diff
@@ -0,0 +1,160 @@
+"""Benchmarks for Gaussian filtering."""
+import numpy as np
+
+import probnum.filtsmooth as pnfs
+import probnum.statespace as pnss
+from probnum.randvars import Normal
+
+
+def load_pendulum():
+    """Set up pendulum problem"""
+
+    np.random.seed(12345)
+
+    delta_t = 0.0075
+    var = 0.32 ** 2
+    g = 9.81
+
+    def f(t, x):
+        x1, x2 = x
+        y1 = x1 + x2 * delta_t
+        y2 = x2 - g * np.sin(x1) * delta_t
+        return np.array([y1, y2])
+
+    def df(t, x):
+        x1, x2 = x
+        y1 = [1, delta_t]
+        y2 = [-g * np.cos(x1) * delta_t, 1]
+        return np.array([y1, y2])
+
+    def h(t, x):
+        x1, x2 = x
+        return np.array([np.sin(x1)])
+
+    def dh(t, x):
+        x1, x2 = x
+        return np.array([[np.cos(x1), 0.0]])
+
+    q = 1.0 * (
+        np.diag(np.array([delta_t ** 3 / 3, delta_t]))
+        + np.diag(np.array([delta_t ** 2 / 2]), 1)
+        + np.diag(np.array([delta_t ** 2 / 2]), -1)
+    )
+    r = var * np.eye(1)
+    initmean = np.ones(2)
+    initcov = var * np.eye(2)
+    dynamod = pnss.DiscreteGaussian(2, 2, f, lambda t: q, df)
+    measmod = pnss.DiscreteGaussian(2, 1, h, lambda t: r, dh)
+    initrv = Normal(initmean, initcov)
+    return dynamod, measmod, initrv, {"dt": delta_t, "tmax": 4}
+
+
+class Filtering:
+    """Benchmark Kalman filter for different linearization techniques."""
+
+    param_names = ["linearization"]
+    params = [["ekf", "ukf"]]
+
+    def setup(self, linearization):
+        dynmod, measmod, initrv, info = load_pendulum()
+        _lin_method = {
```

Comment:
I am curious: is there any reason for the leading underscore? And I, personally, like to split the dict containing options from the actual choosing, i.e. `choose_lin_method = {...}; _lin_method = choose_lin_method[linearization]`. This is super minor, but perhaps your method is better... :)

Reply:
No, there is no reason, other than to mark this variable as not relevant _as_ a variable. It merely serves as a `switch ... case ...` substitute.

Reply:
hmmm that would make it a relevant variable, no? :D I think that this is very subjective, though, so feel free to ignore this comment.

Reply:
I find it ugly as well, and I see no point in keeping it this way. Seemed right at the time, but I'll get rid of it 😄